c7n/resources/__init__.py

```diff
@@ -23,7 +23,9 @@ def load_resources(resource_types=('*',)):
         if '*' in pmap:
             p.get_resource_types(('*',))
         elif pname in pmap:
-            _, not_found = p.get_resource_types(pmap[pname])
+            resource_classes, not_found = p.get_resource_types(pmap[pname])
+            for resource in resource_classes:
+                p.initialize_resource(resource)
```

Comment:
what are the pros/cons of doing this at load time vs at registration time?

Reply:
I'm unsure where else to have put the hook. This is one of the only places that I found that satisfied the ability to see the registrations even when viewing the schema such as `poetry run custodian schema s3.filters`

Reply:
What did you have in mind wrt registration time for loading the default registrations?

I reviewed two possibilities:

* https://github.com/cloud-custodian/cloud-custodian/blob/master/c7n/manager.py#L52-L57
  * I had an issue here where running a `schema` wasn't including the default registrations
  
* https://github.com/cloud-custodian/cloud-custodian/blob/master/c7n/resources/aws.py#L555
  * massaging the default PluginRegistry when it is declared seems like another possible location but that seemed messy
  
I like the load time approach as it was straightforward to understand. Here, we have the context of the provider and as the resources get lazy loaded, so too do the filters and actions get loaded.